### PR TITLE
Improve dev environment

### PR DESCRIPTION
**Pull request by unional**
Running the risk of being brutally dismissed by you 😜 , I did an attempt to give it a try on how using two tsconfig, one for `tsc` and one for `language service`, that would improve the dev environment IMO. 😄 

With this setup, all `*.spec.ts` are properly typed instead of giving error on modules because `tsconfig.json` uses `exclude` instead of `files` so `*.spec.ts` are included in the resolution process.

I also updated `typings.json` so the `custom-typings` is simplified.

### CODE REVIEW

**Comment by blakeembrey** [comment]
Seems mostly reasonable, what is `spec-out` though?

**Comment by unional** [comment]
Ah, that's from the `ava` branch. I shall remove it

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@
 npm-debug.log
 /dist/
 /typings/
+/spec-out/
```
Comment (blakeembrey) [inline .gitignore]: Still in `.gitignore`.

File: custom_typings/node.d.ts [hunk 2]
```diff
@@ -3,9 +3,3 @@
 declare class Intl {
 
 }
```
Comment (blakeembrey) [inline custom_typings/node.d.ts]: I guess we now need to add this interface to make it compile with `noLib`!

**Comment by blakeembrey** [comment]
I think I'd like to attempt to merge `tsconfig.json` and `tsconfig.build.json` together. It'll probably require a couple of tweaks - especially since you can't properly use `noLib` with `exclude`. At least we can hack around that limitation by adding the TypeScript typings with `typings`. Aside from that, we could get rid of most the excludes and use `rootDir` instead?

**Comment by unional** [comment]
Never know that is possible. If that is possible, of course! :smile:

**Comment by blakeembrey** [comment]
@unional Would you prefer https://github.com/typings/core/pull/130?

**Pull request closed without merging**